packages/relay-runtime/store/observeFragmentExperimental.js

```diff
@@ -173,6 +170,67 @@ function observeSelector<TFragmentType: FragmentType, TData>(
   });
 }
 
+function observePluralSelector<
+  TFragmentType: FragmentType,
+  TData: Array<mixed>,
+>(
+  environment: IEnvironment,
+  fragmentNode: Fragment<TFragmentType, TData>,
+  fragmentSelector: PluralReaderSelector,
+): Observable<FragmentState<TData>> {
+  const snapshots = fragmentSelector.selectors.map(selector =>
+    environment.lookup(selector),
+  );
+
+  return Observable.create(sink => {
+    const states = snapshots.map((snapshot, index) =>
+      snapshotToFragmentState(
+        environment,
+        fragmentNode,
+        fragmentSelector.selectors[0].owner,
+        snapshot,
+      ),
+    );
+
+    const reduceState = () =>
+      states.reduce<FragmentState<TData>>(
+        (acc, state) => {
+          if (acc.state !== 'ok') {
+            return acc;
+          }
+
+          if (state.state === 'ok') {
+            acc.value.push(state.value);
+            return acc;
+          } else {
+            return state;
+          }
+        },
+        {
+          state: 'ok',
+          // $FlowFixMe[incompatible-cast]: TData is enforced to be an array
+          value: ([]: TData),
+        },
+      );
+
+    sink.next(reduceState());
+
+    const subscriptions = snapshots.map((snapshot, index) =>
+      environment.subscribe(snapshot, latestSnapshot => {
+        states[index] = snapshotToFragmentState(
+          environment,
+          fragmentNode,
+          fragmentSelector.selectors[0].owner,
+          latestSnapshot,
+        );
+        sink.next(reduceState());
```

Comment:
Maybe a comment here calling out that this does not batch updates, and if a store update impacting multiple items in the list is published, we will notify the subscriber multiple times?

Reply:
https://github.com/facebook/relay/pull/4862/commits/06903bfa7a0030c3d5569051b47479e968627bf8